app/Http/Controllers/DisciplineController.php

```diff
@@ -18,16 +22,26 @@ class DisciplineController extends Controller
      *
      * @return \Illuminate\Http\Response
      */
-    public function index()
-    {
-        // $disciplines = Discipline::select('select * from disciplines');
 
-        //Ajustar isso para quando estiver o campo no banco de dados para a verificação de trailer
-        $disciplines = DB::table('disciplines')
-        ->select('disciplines.*',
-            (DB::raw("(SELECT medias.url FROM medias WHERE medias.discipline_id = disciplines.id and medias.type = 'video' and medias.is_trailer = '1' ) AS urlMedia")))
-        ->get();
+    //Objetos de Model
+    private $objDiscipline;
+    private $objUser;
+    private $objProfessor;
+    private $objMedia;
+    private $objClassificationDiscipline;
 
+    public function __construct()
+    {
+        $this->objDiscipline = new Discipline();
+        $this->objUser = new User();
+        $this->objProfessor = new Professor();
+        $this->objMedia = new Media();
+        $this->objClassificationDiscipline = new ClassificationDiscipline();
```

Comment:
Algum motivo para fazer isso e não usar o model diretamente?

Reply:
Para facilitar o uso do model. Utilizando esse método a seguinte expressao:
        $professor = new User();
        $professor = $professor->find(Auth::id())->professor();
fica dessa forma:
 $professor = $this->objUser->find(Auth::id())->professor();

Reply:
Você também pode fazer da seguinte forma:
```
User::find(Auth::id())
```

Sendo que nesse caso você só tá querendo pegar o professor, então você pode fazer de duas formas:
```
$professor = Auth::user()->professor;
// ou
$professor = Professor::query()->where('user_id', Auth::id())->first();
```